Adjust --detailed-resources to sierra-gas tracking

THenry14 opened the pull request:
Related to #2977

This PR stack aims to add support for new resource tracking (sierra-gas) and new gas representation (GasVector triplet instead of single l1 gas value).

-- (https://github.com/foundry-rs/starknet-foundry/pull/3049) Add sierra gas tracking
-- (https://github.com/foundry-rs/starknet-foundry/pull/3050) Run already existing tests with CairoSteps set explicitly
-- (This PR) Add support for sierra gas in `--detailed-resources`
-- (https://github.com/foundry-rs/starknet-foundry/pull/3052) Add support for sierra gas in `--save-trace-data`
-- (https://github.com/foundry-rs/starknet-foundry/pull/3053) Add sierra version assertion
-- (https://github.com/foundry-rs/starknet-foundry/pull/3054) Introduce new gas representation
-- (https://github.com/foundry-rs/starknet-foundry/pull/3055) Add gas tests for sierra-gas tracking
-- (https://github.com/foundry-rs/starknet-foundry/pull/3056) Update docs with sierra-gas related changes

THenry14 committed:
[skip ci] Adjust --detailed-resources to sierra-gas tracking

cptartur commented on crates/forge/tests/e2e/running.rs:
```diff
@@ -847,6 +847,33 @@ fn detailed_resources_flag() {
 }
 
+#[test]
+#[cfg_attr(not(feature = "scarb_since_2_10"), ignore)]
```
I think this PR should set a CI job as well to run this

THenry14 replied:
This is done in https://github.com/foundry-rs/starknet-foundry/pull/3055 , but if you really want it here I can backport :+1:

cptartur replied:
Not needed, I'll review it there then.

ksew1 commented on crates/forge/tests/e2e/running.rs:
```diff
@@ -847,6 +847,33 @@ fn detailed_resources_flag() {
 }
 
+#[test]
+#[cfg_attr(not(feature = "scarb_since_2_10"), ignore)]
```
What error will the user get if they are running scarb 2.9.* and use these flags?

THenry14 replied:
Right now? It will pass, but exec resources would be 0 (as far as I remember from testing this, might as well panic on something weird like printing/adding/subtracting)

After the stack gets merged? You will get:

```
[ERROR] Tracking SierraGas is not supported for sierra <= 1.7.0: Contract version 1.6.0 is lower than required minimal sierra version
```

As sierra check will fail (added in https://github.com/foundry-rs/starknet-foundry/pull/3053)

ksew1 replied:
Could we add test for that?

THenry14 replied:
Done https://github.com/foundry-rs/starknet-foundry/pull/3051/commits/eb5446323a98153feb205b996fa27082070f0c86

THenry14 committed:
Add additional test for sierra assertion

THenry14 committed:
Add sierra gas resource to trace file (#3052)

Related to #2977

This PR stack aims to add support for new resource tracking (sierra-gas)
and new gas representation (GasVector triplet instead of single l1 gas
value).

-- (https://github.com/foundry-rs/starknet-foundry/pull/3049) Add sierra
gas tracking
-- (https://github.com/foundry-rs/starknet-foundry/pull/3050) Run
already existing tests with CairoSteps set explicitly
-- (https://github.com/foundry-rs/starknet-foundry/pull/3051) Add
support for sierra gas in `--detailed-resources`
--> (This PR) Add support for sierra gas in `--save-trace-data`
-- (https://github.com/foundry-rs/starknet-foundry/pull/3053) Add sierra
version assertion
-- (https://github.com/foundry-rs/starknet-foundry/pull/3054) Introduce
new gas representation
-- (https://github.com/foundry-rs/starknet-foundry/pull/3055) Add gas
tests for sierra-gas tracking
-- (https://github.com/foundry-rs/starknet-foundry/pull/3056) Update
docs with sierra-gas related changes
